bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/client/HomeConnectEventSourceListener.java

```diff
@@ -0,0 +1,254 @@
+                } catch (Exception e) {
+                    logger.error("Could not publish closed event to listener ({})!", haId, e);
+                }
+                stopMonitor();
+            }
+        }, SSE_MONITOR_INITIAL_DELAY, SSE_MONITOR_INTERVAL, TimeUnit.MINUTES);
+    }
+
+    private void stopMonitor() {
+        logger.debug("Dispose event source connection monitor of appliance ({}).", haId);
+        eventSourceMonitorFuture.cancel(true);
+    }
+
+    private List<Event> mapEventSourceEventToEvent(String haId, EventType type, @Nullable String data) {
+        List<Event> events = new ArrayList<>();
+
+        if ((STATUS.equals(type) || EVENT.equals(type) || NOTIFY.equals(type)) && data != null && !isEmpty(data)
+                && !EMPTY_DATA.equals(data)) {
+            try {
+                JsonObject responseObject = jsonParser.parse(data).getAsJsonObject();
+                JsonArray items = responseObject.getAsJsonArray("items");
+
+                items.forEach(item -> {
+                    JsonObject obj = (JsonObject) item;
+                    @Nullable
+                    String key = obj.get("key") != null ? obj.get("key").getAsString() : null;
+                    @Nullable
+                    String value = obj.get("value") != null && !obj.get("value").isJsonNull()
+                            ? obj.get("value").getAsString()
+                            : null;
+                    @Nullable
+                    String unit = obj.get("unit") != null ? obj.get("unit").getAsString() : null;
+                    @Nullable
+                    String name = obj.get("name") != null ? obj.get("name").getAsString() : null;
+                    @Nullable
+                    String uri = obj.get("uri") != null ? obj.get("uri").getAsString() : null;
+                    @Nullable
+                    EventLevel level = obj.get("level") != null
+                            ? EventLevel.valueOfLevel(obj.get("level").getAsString())
+                            : null;
+                    @Nullable
+                    EventHandling handling = obj.get("handling") != null
+                            ? EventHandling.valueOfHandling(obj.get("handling").getAsString())
+                            : null;
+                    @Nullable
+                    Long timestamp = obj.get("timestamp") != null ? obj.get("timestamp").getAsLong() : null;
+                    @Nullable
+                    ZonedDateTime creation = timestamp != null
+                            ? ZonedDateTime.ofInstant(Instant.ofEpochSecond(timestamp),
+                                    TimeZone.getDefault().toZoneId())
+                            : null;
+
+                    events.add(new Event(haId, type, key, name, uri, creation, level, handling, value, unit));
+                });
+            } catch (IllegalStateException e) {
+                logger.error("Could not parse event! haId={}, error={}", haId, e.getMessage());
```

Comment:
Use the implicit last argument to the logging functions to log exceptions:
```java
logger.error("Could not parse event! haId={}", haId, e);
```
(Valid verywhere `e.getMessage() is passed to a logging method)